esp/esdllib/esdl_script.cpp

```diff
@@ -53,26 +53,65 @@ class CEsdlScriptContext : public CInterfaceOf<IEsdlScriptContext>
     virtual IEsdlFunctionRegister* queryFunctionRegister() const override { return functionRegister; }
     virtual void setTraceToStdout(bool val) override
     {
-        traceToStdout = val;
+        if (val != traceToStdout)
+        {
+            traceToStdout = val;
+            tracer->setSink(val ? consoleSink.get() : jlogSink.get());
+        }
     }
     virtual bool getTraceToStdout() const override { return traceToStdout; }
     virtual void setTestMode(bool val) override { testMode = val; }
     virtual bool getTestMode() const override { return testMode; }
+    virtual ITracer& trace() const override { return *tracer; }
+    virtual bool enableMasking(const char* domainId, uint8_t version) override
+    {
+        if (!maskingEngine)
+        {
+            trace().uwarnlog("enable masking request not completed - no masking engine");
+            return false;
+        }
+        if (maskingEnabled())
+        {
+            trace().uwarnlog("enable masking request not completed - already enabled");
```

Comment:
Can you output the current and requested domain and version?  If they match may not even need to warn?